frontend/src/pages/projects/screens/spawner/storage/ClusterStorageTable.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+
+import { Flex, FlexItem, Icon, Popover, Truncate } from '@patternfly/react-core';
+import { Tr, Td, ActionsColumn } from '@patternfly/react-table';
+import { InfoCircleIcon } from '@patternfly/react-icons';
+
+import { Table } from '~/components/table';
+import { StorageData, StorageType } from '~/pages/projects/types';
+import { clusterStorageTableColumns } from './constants';
+import { ClusterStorageEditModal } from './ClusterStorageEditModal';
+import { ClusterStorageDetachModal } from './ClusterStorageDetachModal';
+
+interface ClusterStorageTableProps {
+  storageData: StorageData[];
+  workbenchName: string;
+  setStorageData: React.Dispatch<React.SetStateAction<StorageData[]>>;
+}
+
+export const ClusterStorageTable: React.FC<ClusterStorageTableProps> = ({
+  storageData,
+  workbenchName,
+  setStorageData,
+}) => {
+  const [isEditModalOpen, setIsEditModalOpen] = React.useState(false);
+  const [isDetachModalOpen, setIsDetachModalOpen] = React.useState(false);
+  const [selectedId, setSelectedId] = React.useState<number>();
+  const selectedStorage = storageData.find((formData) => formData.id === selectedId);
+  const hasUpdatedDefaultNameRef = React.useRef(false);
+
+  const updateStorageData = React.useCallback(
+    (newStorageData: StorageData) => {
+      setStorageData((prevData) =>
+        prevData.map((pvcData, pvcIndex) => {
+          if (pvcIndex === selectedId) {
```

Comment:
On line 27, we correctly compare selectedId with the storageData.id. The same comparison logic should be applied here for consistency.

Since `pvcData.id` is the actual unique identifier for the storage data item, it should be matched against selectedId rather than using the array index.

This will ensure that the correct storage data item is updated, regardless of its position in the array.


```suggestion
          if (pvcData.id === selectedId) {
```

Reply:
The position in the array is what the ID is, so it wouldn't matter either way.

Reply:
But if we detach a storageData, the subsequent items shift positions, but their id remains unchanged. In that case wouldn't the index will be different from the id?